mne/preprocessing/_csd.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2003-2010 Jürgen Kayser <rjk23@columbia.edu>
+# Copyright 2017 Federico Raimondo <federaimondo@gmail.com> and
+#                Denis A. Engemann <dengemann@gmail.com>
+#
+#
+# The original CSD Toolbox can be find at
+# http://psychophysiology.cpmc.columbia.edu/Software/CSDtoolbox/
+
+# Authors: Denis A. Engeman <denis.engemann@gmail.com>
+#          Alex Rockhill <aprockhill206@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from scipy.linalg import inv
+
+from mne import pick_types, pick_info
+from mne.utils import logger
+from mne.io import BaseRaw
+from mne.epochs import BaseEpochs
+from mne.evoked import Evoked
+from mne.parallel import parallel_func
+from mne.channels.interpolation import _calc_g, _calc_h
+
+
+def _prepare_G(G, lambda2):
+    G.flat[::len(G) + 1] += lambda2
+    # compute the CSD
+    Gi = inv(G)
+
+    TC = Gi.sum(0)
+    sgi = np.sum(TC)  # compute sum total
+
+    return Gi, TC, sgi
+
+
+def _compute_csd(data, G_precomputed, H, head):
+    """compute the CSD"""
+    n_channels, n_times = data.shape
+    mu = data.mean(0)[None]
+    Z = data - mu
+    X = np.zeros_like(data)
+    head **= 2
+
+    Gi, TC, sgi = G_precomputed
+
+    Cp2 = np.dot(Gi, Z)
+    c02 = np.sum(Cp2, axis=0) / sgi
+    C2 = Cp2 - np.dot(TC[:, None], c02[None, :])
+    X = np.dot(C2.T, H).T / head
+    return X
+
+
+def compute_current_source_density(inst, picks=None, g_matrix=None,
+                                   h_matrix=None, lambda2=1e-5,
+                                   stiffness=4, n_legendre_terms=50,
+                                   head=1.0, n_jobs=1, copy=True):
+    """ Current Source Density (CSD) transformation
+
+    Transformation based on spherical spline surface Laplacian as suggested by
+    Perrin et al. (1989, 1990), published in appendix of Kayser J, Tenke CE,
+    Clin Neurophysiol 2006;117(2):348-368)
+
+    Implementation of algorithms described by Perrin, Pernier, Bertrand, and
+    Echallier in Electroenceph Clin Neurophysiol 1989;72(2):184-187, and
+    Corrigenda EEG 02274 in Electroenceph Clin Neurophysiol 1990;76:565.
+
+    Note: This function applies an average reference to the data.
+          Do not transform CSD data to source space.
+
+    Parameters
+    ----------
+    inst : instance of Raw, Epochs or Evoked
+        The data to be transformed.
+    picks : np.ndarray | list, shape (n_channels,) | None
+        The picks to be used. Defaults to None.
+    lambda2 : float
+        Regularization parameter, produces smoothnes. Defaults to 1e-5.
+    stiffnes : float
+        stiffnes of the spline. Also referred to as `m`
+        (if g_matrix or h_matrix is provided this will be ignored)
+    n_legendre_terms : int
+        number of Legendre terms to evaluate (if g_matrix or h_matrix
+        is provided this will be ignored)
+    head : float
+        The head radius (unit sphere). Defaults to 1.
```

Comment:
This should probably be a `sphere` arg, a tuple containing the center and radius. The default should probably be `(0., 0., 0., 0.095)` to match how we create our montages.

Reply:
I think I need a bit of help implementing this in _compute_csd @larsoner do you have a second?

Reply:
It's just implemented with the radius now, you can add the center shift later or explain how this is done so I can or if you give me a bit more info I can figure it out.

Reply:
In fact it is mere downscaling for the new CSD values will not differ from original microvolts. 
If you want meaningful review then you can introduce `scaling_factor` or like
And if it =None then calculate a head radius from "real" head measurements.
So I suggest to remove this scaling factor from low level calc function to an upper level.